src/content/language/macros.md

```diff
@@ -0,0 +1,287 @@
+  its dependencies.
+2. [Add the experiment][] to the `analysis_options.yaml`
+  file at the root of your project:
+
+   ```yaml
+   analyzer:
+    enable-experiment:
+      - macros
+   ```
+3. Import it in the file you plan to use it:
+
+  ```dart
+  import 'package:json/json.dart';
+  ```
+
+4. Run your project with the experiment flag:
+
+  ```console
+  dart --enable-experiment=macros run bin/my_app.dart
+  ```
+
+[Add the package]: /guides/packages
+[Add the experiment]: /tools/experiment-flags#using-experiment-flags-with-the-dart-analyzer-command-line-and-ide
+
+### Use the macro
+
+To use the `JsonCodable` macro, append the annotation to the class you want to serialize:
+
+```dart
+import 'package:json/json.dart';
+
+@JsonCodable() // Macro annotation.
+class User {
+ final int? age;
+ final String name;
+ final String username;
+}
+```
+
+The macro introspects the `User` class and derives the implementations of
+`fromJson` and `toJson` using the `User` class's fields.
+
+So, without needing to define them yourself, `toJson` and `fromJson` are now
+available to use on objects of the annotated class:
+
+```dart
+void main() {
+ // Given some arbitrary JSON:
+ var userJson = {
+   'age': 5,
+   'name': 'Roger',
+   'username': 'roger1337'
+ };
+
+ // Use the generated members:
+ var user = User.fromJson(userJson);
+ print(user);
+ print(user.toJson());
+}
+```
+
+### View the generated code
+
+You can optionally view the generated code.
+Click on the "**Go to Augmentation**" link that appears under the annotation
+in your IDE (supported in VSCode and IntelliJ)
+to see how the macro generates `toJson` and `fromJson`.
+
+If you change anything in the annotated class, you can watch the generated augmentation
+adjust in real time alongside your application code:
+
+![A side-by-side gif of the generated augmentation updating as the code it's augmenting is updated](/assets/img/language/macro-augmentation.gif)
+
+### Trigger custom diagnostics
+
+The `JsonCodable` macro has built-in diagnostics that are emmitted just like
+diagnostics from the language itself. For example, if you try to manually
+declare a `toJson` method where the macro is applied, the analyzer will emit
+the error:
+
+```dart
+@JsonCodable()
+class HasToJson {
+ void [!toJson!]() {}
+ // Error: Cannot generate a toJson method due to this existing one.
+}
+```
+
+You can search "`DiagnosticMessage`" in the [the definition of `JsonCodable`][json]
+for other errors the macro will throw. For example, extending a class that isn't
+also serializable, or if field names don't exactly match the key names in the given JSON.
+
+:::note
+To learn more about using the `JsonCodable` macro, like supported field types,
+treatment of null and generics, and more, check out [the README][].
+:::
+
+[the definition of `JsonCodable`]: {{site.repo.dart.sdk}}/blob/master/pkg/json/lib/json.dart
+[the README]: {{site.repo.dart.sdk}}/blob/main/pkg/json/README.md
```

Comment:
link to pub here too

Reply:
So, all the content shows up in the Preview version of the pub page, https://pub.dev/packages/json/versions/0.20.0, but by default it lands on Latest. So I've used the full link to 0.20.0 for now, but we'll have to update it